### ISSUE TITLE
Weird behaviour

### ISSUE THREAD

**Issue by easiwriter** [rajdeep/proton#351]
Hi

There is something weird going on. Pressing on the text and dragging sends insertion requests to add s space at the insertion point. This happens in the example app as well as shown in the video.

https://github.com/user-attachments/assets/26c88b5e-6f4f-4ea5-b21d-7a0dd295c5d8

This is using iOS 18.2 and Xcode 16.1.

**Comment by rajdeep** [comment]
weird indeed. not sure what might be needing a fix here. Seems like an iOS bug to me. I'll take a look at this and get back to you in next few days. thanks for highlighting the issue.

**Comment by rajdeep** [comment]
I tried in Xcode 16.2, iOS 18.2 and see no issues on dragging of text insertion pointer:

https://github.com/user-attachments/assets/d55eb01b-d4e2-442e-a047-f0b979e92b9f

**Comment by rajdeep** [comment]
To add new menu item, you can use `buildMenu` available in `UIViewController` like so:

```
 override func buildMenu(with builder: any UIMenuBuilder) {
        super.buildMenu(with: builder)
        let commentMenu = createTestMenu()
        builder.insertChild(commentMenu, atStartOfMenu: .root)
    }

    func createTestMenu() -> UIMenu {
        let commentItem = UICommand(title: "Test", action: #selector(test))
        let commentMenu = UIMenu(title: "Test",
                                 image: nil,
                                 identifier: .application,
                                 options: .displayInline,
                                 children: [])
        return commentMenu
    }

    @objc private func test() {
        print("Test clicked")
    }

```

which shows up new menu item:

![Simulator Screenshot - iPhone 15 Pro - 2024-11-24 at 14 39 50](https://github.com/user-attachments/assets/8d264ba5-07cc-49cd-8907-c1de5cafff85)

**Comment by easiwriter** [comment]
The drag problem is with the IOS18.2 simulator. IOS17.2 is OK. 

Thanks for the menu code.

**Comment by rajdeep** [comment]
The recording I shared is from Xcode 16.2 running iOS 18.2 simulator. Can you try again on Xcode 16.2?

**Comment by easiwriter** [comment]
Just tried in Xcode 16.2 beta 3 and iOS 18.2 sim. It works when entering text into the Commands panel. The problem occurs using the Autogrowing text panel. Also I have installed the main branch of Proton because the next major version doesn't compile (can't find TableView).

**Comment by rajdeep** [comment]
I tried that again on the Autogrowing Example and that seems to work too. Just to confirm, what is the macOS version that you are running? Mine is `14.7 (23H124)` and have not yet updated to `15.0`

https://github.com/user-attachments/assets/39cf48cc-3182-4697-989d-e1bc18e3b5f1

If you are on 15.0, it would be great if you can try a `UITextView` in an sample app and check the behaviour there.

**Comment by easiwriter** [comment]
I've worked out what is going on. If you tap and drag without delay it's fine, but if you tap and hold for a second then drag you get the behaviour. Here is a simple app  and video that shows this.

[EditorTest.zip](https://github.com/user-attachments/files/17968352/EditorTest.zip)

https://github.com/user-attachments/assets/43ee368a-1c0c-4916-bd5f-cb30f12722af

**Comment by rajdeep** [comment]
@easiwriter, I am still not seeing the issue. Can you confirm you are using **Xcode 16.2 beta 3/ iOS 18.2**? If you are seeing this issue in `TextEditor` as well, I am afraid it's something to do with Apple and I won't be able to fix that. I would recommend creating a Bug with Apple in case you are seeing this issue on latest beta of Xcode.

https://github.com/user-attachments/assets/e06be089-6ecf-4039-86dc-aaca9d4aaa1e

**Comment by easiwriter** [comment]
Hi, I am using Xcode 16.2 beta 3/ IOS 18.2 although it does happen on earlier versions. I am also using Sequoia 15.1.1. 

The fact that the sample only uses TextEditor shows that this has to be an Apple issue and nothing to do with Proton so I am submitting a Feedback request. 

Cheers
Keith

**Comment by easiwriter** [comment]
I submitted a feedback request and followed up on their response. It is a bug fixed in IOS RC 18.2. I've tested it and it is OK.

**Comment by rajdeep** [comment]
that's great. thanks for the update, @easiwriter
